core/connection/session.go

```diff
@@ -25,14 +25,43 @@ import (
 	"github.com/mysteriumnetwork/node/session"
 )
 
+// SessionStatus represents list of possible session statuses
+type SessionStatus int
+
+const (
+	// SessionStatusNew means that newly created session object is written to storage
+	SessionStatusNew = SessionStatus(0)
+	// SessionStatusCompleted means that session object is updated on connection disconnect event
+	SessionStatusCompleted = SessionStatus(1)
+)
+
 // Session holds structure for saving session history
 type Session struct {
 	SessionID       session.ID `storm:"id"`
 	ProviderID      identity.Identity
 	ServiceType     string
 	ProviderCountry string
 	TimeStarted     time.Time
-	// these will be updated while session is ongoing
-	Duration  int // in seconds
-	DataStats stats.SessionStats
+	Status          SessionStatus
+	TimeUpdated     time.Time          // is updated on disconnect event
+	DataStats       stats.SessionStats // is updated on disconnect event
+}
+
+// GetDuration returns delta in seconds (TimeUpdated - TimeStarted)
+func (se *Session) GetDuration() uint64 {
+	if se.TimeUpdated.IsZero() {
+		return 0
```

Comment:
If a session is not finished, `GetDuration` can return `time.Now() - se.TimeStarted` instead of 0.

Reply:
If a session is not finished, that can mean a case where app crashes. So in that case `time.Now() - se.TimeStarted` will provide misleading result.
In current logic, in that case it will provide value 0.